test/index.test.js

```diff
@@ -180,7 +180,30 @@ describe('index', () => {
                 .then(() => assert.fail('Should not get here'))
                 .catch((error) => {
                     assert.calledWith(requestMock, expectedOptions);
-                    assert.match(error.message, 'STATUS CODE 404');
+                    assert.match(error.message, 'Cannot find repository');
+                });
+        });
+
+        it('rejects if status code is not 200 & 404', () => {
+            fakeResponse = {
+                statusCode: 500,
+                body: {
+                    error: {
+                        message: 'INternal Server Error'
+                    }
+                }
+            };
+
+            requestMock.yieldsAsync(null, fakeResponse, fakeResponse.body);
+
+            return scm.parseUrl({
+                checkoutUrl: 'https://batman@bitbucket.org/batman/test.git#mynewbranch',
+                token: 'myAccessToken'
+            })
+                .then(() => assert.fail('Should not get here'))
```

Comment:
It's confusing this way too. If the `assert.fail` is hit, it'll be caught by the next `.catch` in the chain. If you're lucky, you'll have an error that says "'Should not get here' does not equal 'STATUS CODE 500'". And that's assuming mocha unrolls your unit test error message